cms/static/cms/js/modules/cms.base.js

```diff
@@ -12,231 +12,231 @@ window.Class = window.Class || undefined;
-			if(settings === null) settings = this.setSettings(CMS.config.settings);
-
-			// save settings
-			CMS.settings = settings;
-
-			// ensure new settings are returned
-			return CMS.settings;
-		}
-
-	};
-
-	// autoinits
-	CMS.API.Helpers.preventSubmit();
+    /*!
+     * CNS.API.Helpers
+     * Multiple helpers used accross all CMS features
+     */
+    CMS.API.Helpers = {
+
+        // redirects to a specific url or reloads browser
+        reloadBrowser: function (url, timeout, ajax) {
+            var that = this;
+            // is there a parent window?
+            var parent = (window.parent) ? window.parent : window;
+
+            // if there is an ajax reload, prioritize
+            if(ajax) {
+                parent.CMS.API.locked = true;
+                // check if the url has changed, if true redirect to the new path
+                // this requires an ajax request
+                $.ajax({
+                    'async': false,
+                    'type': 'GET',
+                    'url': parent.CMS.config.request.url,
+                    'data': {
+                        'model': parent.CMS.config.request.model,
+                        'pk': parent.CMS.config.request.pk
+                    },
+                    'success': function (response) {
+                        parent.CMS.API.locked = false;
+
+                        if(response === '' && !url) {
+                            // cancel if response is empty
+                            return false;
+                        } else if(parent.location.pathname !== response) {
+                            // api call to the backend to check if the current path is still the same
+                            that.reloadBrowser(response);
+                        } else if(url === 'REFRESH_PAGE') {
+                            // if on_close provides REFRESH_PAGE, only do a reload
+                            that.reloadBrowser();
+                        } else if(url) {
+                            // on_close can also provide a url, reload to the new destination
+                            that.reloadBrowser(url);
+                        }
+                    }
+                });
+
+                // cancel further operations
+                return false;
+            }
+
+            // add timeout if provided
+            parent.setTimeout(function () {
+                if (url && url != parent.location.href) {
+                    // location.reload() takes precedence over this, so we
+                    // don't want to reload the page if we need a redirect
+                    parent.location.href = url;
+                }
+                else {
+                    // ensure page is always reloaded #3413
+                    parent.location.reload()
+                }
+            }, timeout || 0);
+        },
+
+        // disable multiple form submissions
+        preventSubmit: function () {
+            var forms = $('#cms-toolbar').find('form');
```

Comment:
sadly not currently @darioalbanesi as there are various dependencies affected - but it's a goal